packages/viewer/src/lib/interactions/interactionApi.ts

```diff
@@ -460,16 +465,87 @@ export abstract class InteractionApi<T extends Camera = Camera> {
    *  values zoom out.
    */
   public async zoomCamera(delta: number): Promise<void> {
-    return this.transformCamera(({ camera, viewport }) => {
-      const vv = camera.viewVector;
-      const v = Vector3.normalize(vv);
+    return this.transformCamera(({ camera, viewport, frame }) => {
+      if (viewport != null && frame != null) {
+        const isPerspective = camera?.toFrameCamera().isPerspective();
+
+        if (isPerspective) {
+          const vv = camera.viewVector;
+
+          // Calculate the scalar determining the amount to zoom
+          const distance = Vector3.magnitude(vv);
+          const epsilon = (3 * distance * delta) / viewport.height;
+
+          // Scale the current viewVector by the scalar calculated above to determine how to adjust the camera position
+          const v = Vector3.normalize(vv);
+          const positionChange = Vector3.scale(epsilon, v);
+
+          // Calculate the new camera position
+          const position = Vector3.add(camera.position, positionChange);
+
+          // Update the camera with the new position
+          const newCamera = camera.update({ position });
+          return newCamera;
+        } else {
+          const orthographicCamera = camera as unknown as OrthographicCamera;
+
+          // Calculate the scalar determining the amount to zoom
+          const relativeDelta =
+            3 * (orthographicCamera.fovHeight / viewport.height) * delta;
```

Comment:
What units is this number in?  Are `fovHeight` and `viewport.height` in the same units?

Reply:
Ah, are `viewport.height` and `delta` in the same units?  The location of the parens might be throwing me off...

Reply:
Good point. I updated the parenthesis and added another comment.